doc/subsystems/networking/net-app-api.rst

```diff
@@ -0,0 +1,207 @@
+  applications. This peer address is typically used as a parameter when
+  calling cpp:func:`net_app_connect()`.
+
+* :option:`CONFIG_NET_APP_MY_IPV4_ADDR` option sets a static IPv4 address for
+  the system. This is typically useful only in device testing as normally the
+  system should use DHCPv4 to discover the IPv4 address.
+
+* :option:`CONFIG_NET_APP_PEER_IPV4_ADDR` option specifies what is the peer
+  device IPv4 address. This is only useful when testing client/server type
+  applications. This peer address is typically used as a parameter when
+  connecting to other device.
+
+The following options are only available if IEEE 802.15.4 wireless network
+technology support is enabled.
+
+* :option:`CONFIG_NET_APP_IEEE802154_DEV_NAME` specifies the name of the
+  IEEE 802.15.4 device.
+
+* :option:`CONFIG_NET_APP_IEEE802154_PAN_ID` specifies the used PAN identifier.
+  Note that the PAN id can be changed at runtime if needed.
+
+* :option:`CONFIG_NET_APP_IEEE802154_CHANNEL` specifies the used radio channel.
+  Note that the used channel can be changed at runtime if needed.
+
+* :option:`CONFIG_NET_APP_IEEE802154_RADIO_TX_POWER` specifies the initial
+  radio TX power level. The TX power level can be changed at runtime if needed.
+
+* :option:`CONFIG_NET_APP_IEEE802154_SECURITY_KEY` specifies the initially used
+  security key. The security key can be changed at runtime if needed.
+
+* :option:`CONFIG_NET_APP_IEEE802154_SECURITY_KEY_MODE` specifies the initially
+  used security key mode. The security key mode can be changed at runtime if
+  needed.
+
+* :option:`CONFIG_NET_APP_IEEE802154_SECURITY_LEVEL` specifies the initially
+  used security level. The used security level can be changed at runtime if
+  needed.
+
+Client / Server Applications
+############################
+
+The net-app API provides functions that enable the application to create
+client / server applications easily. If needed, the applications can
+have the communication secured by TLS (for TCP connections) or DTLS (for
+UDP connections) automatically.
+
+A simple **TCP server** application would make the following net-app API
+function calls:
+
+* cpp:func:`net_app_init_tcp_server()` to configure a local address and TCP
+  port.
+
+* cpp:func:`net_app_set_cb()` to configure callback functions to invoke in
+  response to events, such as data reception.
+
+* cpp:func:`net_app_server_tls()` will optionally setup the system for secured
+  connections. To enable the TLS server, also call the
+  cpp:func:`net_app_server_tls_enable()` function.
+
+* cpp:func:`net_app_listen()` will start listening for new client connections.
+
+Creating a **UDP server** is also very easy:
+
+* cpp:func:`net_app_init_udp_server()` to configure a local address and UDP
+  port.
+
+* cpp:func:`net_app_set_cb()` to configure callback functions to invoke in
+  response to events, such as data reception.
+
+* cpp:func:`net_app_server_tls()` will optionally setup the system for secured
```

Comment:
Yes, from application point of view TLS and DTLS are basically the same thing. Internally, the net-app API will need to do some extra things like setting up the cookie for DTLS, but the user API can stay the same.